lib/orcid/identifier_type_mapper.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Orcid
+  # Maps identifier types between ORCID and SUL-PUB.
+  class IdentifierTypeMapper
+    ORCID_ID_TYPE_TO_SUL_PUB_ID_TYPE = {
+      'pmid' => 'PMID',
+      'wosuid' => 'WosUID'
+    }.freeze
+
+    # From https://pub.orcid.org/v3.0/identifiers
+    ORCID_ID_TYPES = %w[
+      agr
+      ark
+      arxiv
+      asin
+      asin-tld
+      authenticusid
+      bibcode
+      cba
+      cienciaiul
+      cit
+      ctx
+      dnb
+      doi
+      eid
+      ethos
+      grant_number
+      hal
+      handle
+      hir
+      isbn
+      issn
+      jfm
+      jstor
+      kuid
+      lccn
+      lensid
+      mr
+      oclc
+      ol
+      osti
+      other-id
+      pat
+      pdb
+      pmc
+      pmid
+      ppr
+      proposal-id
+      rfc
+      rrid
+      source-work-id
+      ssrn
+      uri
+      urn
+      wosuid
+      zbl
+    ].freeze
+
+    # @return [String] ORCID id type or nil if no matching
+    def self.to_orcid_id_type(sul_pub_id_type)
+      id_type = ORCID_ID_TYPE_TO_SUL_PUB_ID_TYPE.invert.fetch(sul_pub_id_type, sul_pub_id_type)
```

Comment:
i'm not quite sure i understand what is happening here... and is there a reason doi is not in the `ORCID_ID_TYPE_TO_SUL_PUB_ID_TYPE` list?

Reply:
never mind, i think i got what is happening